Assets/MixedRealityToolkit.Tests/PlayModeTests/SpeechTests.cs

```diff
@@ -0,0 +1,104 @@
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            PlayModeTestUtilities.TearDown();
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestToggleProfilerCommand()
+        {
+            // Confirm that the Diagnostics system is enabled and the VisualProfiler is in the scene.
+            IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+            MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+            Assert.IsNotNull(diagnosticsSystem, "The diagnostics system is not enabled in the scene.");
+            yield return null;
+
+            // Verfiy that the VisualProfiler is enabled.
+            Assert.IsTrue(diagnosticsSystem.ShowProfiler, "The VisualProfiler is not active.");
+            yield return null;
+
+            // Toggle the profiler visualization off.
+            KeyboardEvent(VirtualKey_9, ScanCode_9, 0, IntPtr.Zero); // key down
+            yield return null;
+            KeyboardEvent(VirtualKey_9, ScanCode_9, KeyboardFlag_KeyUp, IntPtr.Zero); // key up
+            for (int i = 0; i < 10; i++) { yield return null; }
+
+            // Verify that the VisualProfiler is disabled.
+            Assert.IsFalse(diagnosticsSystem.ShowProfiler, "The VisualProfiler is active (should be inactive).");
+            yield return null;
+
+            // Toggle the profiler visualization on.
+            KeyboardEvent(VirtualKey_9, ScanCode_9, 0, IntPtr.Zero); // key down
+            yield return null;
+            KeyboardEvent(VirtualKey_9, ScanCode_9, KeyboardFlag_KeyUp, IntPtr.Zero); // key up
+            for (int i = 0; i < 10; i++) { yield return null; }
+
+            // Verfiy that the VisualProfiler is enabled.
+            Assert.IsTrue(diagnosticsSystem.ShowProfiler, "The VisualProfiler is inactive (should be active).");
+            yield return null;
+        }
+
+        /// <summary>
+        /// Virutal Key Code and Scan Code values used in this test.
+        /// </summary>
+        private const Byte VirtualKey_9 = 0x39;
+        private const Byte ScanCode_9 = 0x0A;
+
+        /// <summary>
+        /// Flags, for the keybd_event function, used in this test.
+        /// </summary>
+        private const UInt32 KeyboardFlag_KeyUp = 0x0002;
+
+        /// <summary>
+        /// P/Invoke signature for the Windows keybd_event function.
+        /// </summary>
+        /// <param name="keyCode">Virtual Key Code for the desired keyboard event.</param>
+        /// <param name="scanCode">Scan Code for the desired keybaord event.</param>
+        /// <param name="flags">Flags used to interpret the keyboard event.</param>
+        /// <param name="extraInfo">Additional information associated with the keyboard event.</param>
+        [DllImport("user32.dll", EntryPoint = "keybd_event")]
+        private static extern void KeyboardEvent(
```

Comment:
Because this is a windows specific binary, we might want to wrap this entire code on a platform specific check so that when people are doing other platforms (i.e. running this test on mac) it doesn't fail expectedly (i.e. user32.dll doesn't exist there)

Reply:
good call